src/include/network/network_connection.h

```diff
@@ -50,28 +51,33 @@ namespace network {
 class NetworkConnection {
  public:
   int thread_id;
-  int sock_fd;                    // socket file descriptor
-  struct event *network_event = nullptr;  // something to read from network
-  struct event *workpool_event = nullptr; // worker thread done the job
-  short event_flags;              // event flags mask
-
-  SSL* conn_SSL_context = nullptr;          // SSL context for the connection
-
-  NetworkThread *thread;          // reference to the network thread
-  std::unique_ptr<ProtocolHandler> protocol_handler_;       // Stores state for this socket
+  int sock_fd;                             // socket file descriptor
+  struct event *network_event = nullptr;   // something to read from network
+  struct event *workpool_event = nullptr;  // worker thread done the job
+  struct event *logpool_event = nullptr;   // logger thread done the job
+  short event_flags;                       // event flags mask
+  bool is_logging_ = false;
+
+  SSL *conn_SSL_context = nullptr;  // SSL context for the connection
+
+  NetworkThread *thread;  // reference to the network thread
+  std::unique_ptr<ProtocolHandler>
+      protocol_handler_;                      // Stores state for this socket
   ConnState state = ConnState::CONN_INVALID;  // Initial state of connection
   tcop::TrafficCop traffic_cop_;
+
+  logging::WalLogManager log_manager_;
```

Comment:
Can we make log_manager a member of traffic_cop_? It seems that it is always passing way down to traffic_cop. We can have a setter function to bind logging event with log_manager.

Reply:
Done